datajoint/expression.py

```diff
@@ -439,6 +439,14 @@ def tail(self, limit=25, **fetch_kwargs):
 
     def __len__(self):
         """:return: number of elements in the result set e.g. ``len(q1)``."""
+        try:
+            if self._left[0]:
+                return self.connection.query(
+                    'SELECT distinct count(*) FROM {from_}{where}'.format(
+                        from_=self.from_clause(),
+                        where=self.where_clause())).fetchone()[0]
+        except IndexError:
```

Comment:
The exception does not seem like the most expressive way to check whether there are joins or not.